docs/tutorials/stateful-contract.md

```diff
@@ -61,30 +61,33 @@ The entry method mainly does two things:
 * Increment the property `count`: 
 
 ```js
-this.counter++;
+this.count++;
 ```
 
-* Validates this update has been correctly recorded into the `txPreimage`, or in another word, the new state of `counter` has been serialized into current tx by calling:
+* Validates this update has been correctly recorded into the contract, or in another word, the new state of `count` has been serialized into current tx by calling:
 
 ```ts
 assert(this.ctx.hashOutputs == hash256(this.buildStateOutput(this.ctx.utxo.value)));
 ```
 
+In the above code, `this.ctx` accesses [ScriptContext](../getting-started/what-is-scriptcontext.md), which contains the entire transaction data.
+
+
 Finally we can get a complete stateful contract `Counter` as below:
 
 ```ts
 export class Counter extends SmartContract {
   @prop(true)
-  counter: bigint;
```

Comment:
Just change this prop to count. Do not change anywhere else.

Reply:
fixed